cassiopeia/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java

```diff
@@ -350,4 +345,42 @@ static String getTesseractProg() {
         return System.getProperty("os.name").startsWith("Windows") ? "tesseract.exe" : "tesseract";
     }
 
+    /**
+     * This function gets the list of available languages of Tesseract.
+     * 
+     * @param propertyManager
+     * @return String[] - the list of available languages
+     */
+    public String[] getTessLangs(IPropertiesManager propertyManager) {
+        String tesseractBin = propertyManager.getProperty(Properties.TESSERACT_BIN_FOLDER);
+        String command = tesseractBin + "/tesseract --list-langs";
+        String output = "";
+        try {
+            Process proc;
+            BufferedReader reader;
+            proc = Runtime.getRuntime().exec(command);
+            reader = new BufferedReader(new InputStreamReader(proc.getInputStream()));
+            String line = "";
+            while ((line = reader.readLine()) != null) {
+                output = output + line + " ";
+            }
+            proc.waitFor();
+            reader.close();
+        } catch (IOException e) {
+            sysMsgHandler.handleMessage("Error while getting Tesserract languages.", e, MessageType.ERROR);
+        } catch (InterruptedException e) {
+            sysMsgHandler.handleMessage("Error while getting Tesserract languages.", e, MessageType.ERROR);
+        }
+        String[] lang_list;
+        // output will contain string such as "List of available languages : ara eng
+        // osd".
+        // Split is used to collect just the language names from output.
+        lang_list = output.split(":");
+
+        String[] languages;
+        languages = lang_list[1].split(" ");
```

Comment:
why don't you read the the output into a list right away since it's one language per line, isn't it?

Reply:
no string manipulations necessary